modules/platforms/cpp/schema/BinaryTupleParser.cpp

```diff
@@ -18,22 +18,61 @@
 #include "BinaryTupleParser.h"
 #include "common/Platform.h"
 
+#include "common/Bytes.h"
+
 #include <cassert>
 #include <cstring>
 #include <stdexcept>
 
+namespace ignite {
+
 namespace {
 
 template <typename T>
-T as(ignite::BytesView bytes) noexcept {
-    T value;
-    std::memcpy(&value, bytes.data(), sizeof(T));
-    return value;
+T loadBytesAs(BytesView bytes, std::size_t offset = 0) noexcept {
+    return bytes::load<Endian::LITTLE, T>(bytes.data() + offset);
 }
 
-} // namespace
+Date loadBytesAsDate(ignite::BytesView bytes) {
+    std::int32_t date = loadBytesAs<std::uint16_t>(bytes);
+    date |= std::int32_t(loadBytesAs<std::int8_t>(bytes, 2)) << 16;
 
-namespace ignite {
+    std::int32_t day = date & 31;
+    std::int32_t month = (date >> 5) & 15;
+    std::int32_t year = (date >> 9); // Sign matters.
+
+    return Date(year, month, day);
+}
+
+Time loadBytesAsTime(ignite::BytesView bytes) {
+    uint64_t time = loadBytesAs<std::uint32_t>(bytes);
+
+    int nano;
```

Comment:
Why is it `int`? Can it be negative? And why not to specify size which is enough to fit the max value (1'000'000'000 I believe)?